### PR TITLE
Change key `log` to UNRESERVED_KEYWORD

### PR DESCRIPTION
**Pull request by fanfuxiaoran**
pgaudit uses `log` as a guc name: pguadit.log, but gpdb sets it as `TYPE_FUNC_NAME_KEYWORD` which means it is a reserved keyword except for function and type name. So when running `set pgaudit.log`, error like `syntax error` will be reported.  Personally, it doesn't make sense. So change it to UNRESERVED_KEYWORD to make pgaudit.log work.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/include/parser/kwlist.h [hunk 1]
```diff
@@ -270,7 +270,7 @@ PG_KEYWORD("localtimestamp", LOCALTIMESTAMP, RESERVED_KEYWORD)
 PG_KEYWORD("location", LOCATION, UNRESERVED_KEYWORD)
 PG_KEYWORD("lock", LOCK_P, UNRESERVED_KEYWORD)
 PG_KEYWORD("locked", LOCKED, UNRESERVED_KEYWORD)
-PG_KEYWORD("log", LOG_P, TYPE_FUNC_NAME_KEYWORD)
+PG_KEYWORD("log", LOG_P, UNRESERVED_KEYWORD)
```
Comment (beeender) [inline src/include/parser/kwlist.h:273]: Does the `gram.y` need to be changed accordingly?

```
type_func_name_keyword:
...
			| LOG_P

```

@higuoxing may have better knowledge on this.

Comment (fanfuxiaoran) [reply src/include/parser/kwlist.h:273]: > Does the gram.y need to be changed accordingly?

Yes, but I forgot about that.  Created another commit to add the modification of gram.y.

**Review by higuoxing — APPROVED** [review]
LGTM!

**Review by beeender — APPROVED** [review]

**Review by adam8157 — APPROVED** [review]

**Review by adam8157 — COMMENTED** [review]
It was there at least eight years ago, not only for `LOG ERRORS INTO`, but I didn't find it in https://www.postgresql.org/docs/current/sql-createtype.html or https://www.postgresql.org/docs/current/sql-createfunction.html

**Comment by fanfuxiaoran** [comment]
> It's also in `PartitionIdentKeyword` section of `gram.y`, please double check.

Thanks for pointing it out. I have checked it,  `PartitionIdentKeyword` is used for the partition columns, but I didn't find that the key word `log` can work on a column like the `hash` does.  Please correct me if I misunderstood.

**Comment by adam8157** [comment]
> > It's also in `PartitionIdentKeyword` section of `gram.y`, please double check.
> 
> Thanks for pointing it out. I have checked it, `PartitionIdentKeyword` is used for the partition columns, but I didn't find that the key word `log` can work on a column like the `hash` does. Please correct me if I misunderstood.

These words in the `PartitionIdentKeyword` list could be used for SELECT element aliases that don't have the AS keyword.

**Review by adam8157 — APPROVED** [review]
Reviewed the history of gram.y, I think the current changes, which remove it from type_func_name_keyword only, are fine.

**Comment by fanfuxiaoran** [comment]
Thanks for your reviews, merged!

**Pull request merged**